### PR TITLE
[CORE-1466] Do not delete projects with repos or pipelines unless forced

### PR DESCRIPTION
**Pull request by robert-uhl**

### CODE REVIEW

File: src/server/pfs/server/driver.go [hunk 1]
```diff
@@ -507,13 +507,30 @@ func (d *driver) listProject(ctx context.Context, cb func(*pfs.ProjectInfo) erro
...
+func (d *driver) deleteProject(ctx context.Context, txnCtx *txncontext.TransactionContext, project *pfs.Project, force bool) error {
 	if err := project.ValidateName(); err != nil {
 		return errors.Wrap(err, "invalid project name")
 	}
 	if err := d.env.AuthServer.CheckProjectIsAuthorizedInTransaction(txnCtx, project, auth.Permission_PROJECT_DELETE, auth.Permission_PROJECT_MODIFY_BINDINGS); err != nil {
 		return errors.Wrapf(err, "user is not authorized to delete project %q", project)
 	}
+
+	var repoInfos []*pfs.RepoInfo
+	if err := d.listRepo(ctx, false /* includeAuth */, "" /* repoType */, []*pfs.Project{project} /* projectsFilter */, func(repoInfo *pfs.RepoInfo) error {
```
Comment (albscui) [inline src/server/pfs/server/driver.go:519]: Nice C-style comments for these naked params 👍

File: src/server/pfs/server/testing/server_test.go [hunk 2]
```diff
@@ -1178,6 +1178,50 @@ func TestPFS(suite *testing.T) {
...
+		env := realenv.NewRealEnv(ctx, t, dockertestenv.NewTestDBConfig(t))
+
+		_, err := env.PachClient.PfsAPIClient.CreateProject(ctx, &pfs.CreateProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.CreateRepo(ctx, &pfs.CreateRepoRequest{Repo: &pfs.Repo{Project: &pfs.Project{Name: "default"}, Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.CreateRepo(ctx, &pfs.CreateRepoRequest{Repo: &pfs.Repo{Project: &pfs.Project{Name: "test"}, Name: "test"}})
+		require.NoError(t, err)
+		_, err = env.PachClient.PfsAPIClient.DeleteProject(ctx, &pfs.DeleteProjectRequest{Project: &pfs.Project{Name: "test"}})
+		require.YesError(t, err)
```
Comment (albscui) [inline src/server/pfs/server/testing/server_test.go:1204]: Maybe also verify the error message with something like 
```Go
require.ErrorContains(t, err, "project test contains 2 repos")
```

**Review by albscui — APPROVED** [review]
LGTM!

File: src/server/pfs/server/driver.go (hunk 1 above)
Comment (robert-uhl) [reply src/server/pfs/server/driver.go:519]: I think that I stole it from a line of code you wrote, to be honest.

File: src/server/pfs/server/testing/server_test.go (hunk 2 above)
Comment (robert-uhl) [reply src/server/pfs/server/testing/server_test.go:1204]: I think that I’d prefer not to hardcode that at the moment. Longer term I think that we need to attack structured errors across RPCs, but … that’s another story.

**Pull request merged**